File: frontend/src/api/agent.ts
// Agent API (P1) - /api/v1/agent/*
//
// 注意：现有 client.ts 的 BASE_URL 是 /api/v1，但 agent 端点走 POST + 命名 SSE 事件，
// 且 createSSE 用的 EventSource 只支持 GET + 默认 message 事件，故这里独立封装
// fetch + ReadableStream 手动解析 SSE。

const BASE = '/api/v1/agent'

async function errorText(resp: Response): Promise<string> {
  try {
    const j = await resp.json()
    return (j && (j.error || j.message)) || resp.statusText
  } catch {
    return resp.statusText
  }
}

export interface CreateSessionResponse {
  session_id: string
}
export interface ToolMeta {
  name: string
  description: string
  input_schema: unknown
}
export interface ListToolsResponse {
  tools: ToolMeta[]
}
export interface ExecuteToolResponse {
  name: string
  result: unknown
}

export interface ChatMessage {
  role: 'user' | 'assistant' | 'tool'
  content: string
  created_at?: string
}

export interface AgentSession {
  id: string
  project_id: string
  title: string | null
  messages: ChatMessage[]
  current_step: string
  created_at: string
  updated_at: string
}

export interface PromptView {
  scope: string
  /** 当前生效提示词（自定义或内置默认） */
  system_prompt: string
  /** 内置默认基座（用于「恢复默认」） */
  default_prompt: string
  /** 是否存在用户自定义覆盖 */
  is_customized: boolean
}

export async function createSession(projectId: string): Promise<CreateSessionResponse> {
  const resp = await fetch(BASE + '/session', {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ project_id: projectId }),
  })
  if (!resp.ok) throw new Error(await errorText(resp))
  return resp.json()
}

export async function getSession(id: string): Promise<AgentSession> {
  const resp = await fetch(BASE + '/session/' + id)
  if (!resp.ok) throw new Error(await errorText(resp))
  return resp.json()
}

/** 列出某项目下的会话（历史侧栏，按项目隔离）。 */
export async function listSessions(projectId: string): Promise<AgentSession[]> {
  const resp = await fetch(BASE + '/sessions?project_id=' + encodeURIComponent(projectId))
  if (!resp.ok) throw new Error(await errorText(resp))
  return resp.json()
}

/** 删除会话（含消息）。 */
export async function deleteSession(id: string): Promise<void> {
  const resp = await fetch(BASE + '/session/' + id, { method: 'DELETE' })
  if (!resp.ok) throw new Error(await errorText(resp))
}

/** 重命名会话。 */
export async function renameSession(id: string, title: string): Promise<void> {
  const resp = await fetch(BASE + '/session/' + id, {
    method: 'PUT',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ title }),
  })
  if (!resp.ok) throw new Error(await errorText(resp))
}

export async function listTools(): Promise<ListToolsResponse> {
  const resp = await fetch(BASE + '/tools')
  if (!resp.ok) throw new Error(await errorText(resp))
  return resp.json()
}

export async function executeTool(
  name: string,
  input: unknown,
  projectId: string,
): Promise<ExecuteToolResponse> {
  const resp = await fetch(BASE + '/tool/execute', {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ name, input, project_id: projectId }),
  })
  if (!resp.ok) throw new Error(await errorText(resp))
  return resp.json()
}

/** confirm_step 的返回结构（来自后端 guide::ValidationReport 的 JSON 序列化） */
export interface ConfirmStepResult {
  passed: boolean
  current_step: string
  current_title: string
  next_step?: string
  next_title?: string
  missing?: Array<{ kind: string; detail: string }>
}

/** 推进当前引导阶段到下一步（用户在前端点"确认推进"按钮触发）。
 *  `targetStep` 可选：传入则跳到指定 step（血肉小选择器用）。
 */
export async function confirmGuideStep(
  projectId: string,
  targetStep?: string,
): Promise<ConfirmStepResult> {
  const resp = await fetch(BASE + '/guide/confirm', {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ project_id: projectId, target_step: targetStep }),
  })
  if (!resp.ok) throw new Error(await errorText(resp))
  return resp.json()
}

/** 引导步骤的就绪情况（`GET /projects/{id}/guide/status`，后端真源）。 */
export interface GuideStepStatus {
  key: string
  title: string
  group: 'skeleton' | 'flesh'
  /** current = 当前所处阶段；complete = 产物已就绪；pending = 还缺产物 */
  status: 'current' | 'complete' | 'pending'
  /** 本步产物是否已满足——步骤条的对勾只看它 */
  ready: boolean
  next: string | null
  missing: Array<{ kind: string; detail: string }>
}

export interface GuideStatus {
  current_step: string
  current_title: string | null
  /** 当前阶段是否已达到推进条件 */
  current_ready: boolean
  steps: GuideStepStatus[]
}

/** 读取项目引导进度。
 *
 *  步骤条的对勾、每步的分组与标题、推进按钮的"还差什么"全部取自这里：
 *  与 `confirm_step` 的推进校验用的是后端同一套快照与同一个 `validate_step`。
 *  **不要在前端另行数数量判断完成度**——那会多出第二份判定标准，
 *  "副线"就曾因此永远不打勾（前端漏了这一步，且没有任何报错）。
 */
export async function getGuideStatus(projectId: string): Promise<GuideStatus> {
  const resp = await fetch(`/api/v1/projects/${projectId}/guide/status`)
  if (!resp.ok) throw new Error(await errorText(resp))
  return resp.json()
}

/** 读取当前生效提示词视图（含内置默认与是否自定义）。 */
export async function getPrompt(scope = 'global'): Promise<PromptView> {
  const resp = await fetch(`${BASE}/prompt?scope=${encodeURIComponent(scope)}`)
  if (!resp.ok) throw new Error(await errorText(resp))
  return resp.json()
}

/** 保存（upsert）自定义提示词基座。 */
export async function savePrompt(systemPrompt: string, scope = 'global'): Promise<PromptView> {
  const resp = await fetch(`${BASE}/prompt`, {
    method: 'PUT',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ scope, system_prompt: systemPrompt }),
  })
  if (!resp.ok) throw new Error(await errorText(resp))
  return resp.json()
}

/** 删除自定义覆盖（恢复内置默认）。 */
export async function deletePrompt(scope = 'global'): Promise<void> {
  const resp = await fetch(`${BASE}/prompt?scope=${encodeURIComponent(scope)}`, {
    method: 'DELETE',
  })
  if (!resp.ok) throw new Error(await errorText(resp))
}

/** 本轮调用的用量统计（含提示缓存命中）。 */
export interface LlmUsage {
  prompt_tokens: number
  completion_tokens: number
  total_tokens: number
  /** 命中提示缓存的 prompt token 数；网关未提供该信息时为 null。 */
  cached_tokens: number | null
  /** 缓存命中率 0~1；网关未提供缓存信息时为 null。 */
  cache_hit_rate: number | null
}

export interface ChatStreamHandlers {
  onStatus?: (data: string) => void
  onToken?: (data: string) => void
  onDone?: () => void
  onError?: (data: string) => void
  onQuestion?: (data: { question: string; options: string[] }) => void
  onTool?: (data: { name: string; input: unknown; ok: boolean; output: string }) => void
  /** 本轮用量统计（在 done 之前下发一次）。 */
  onUsage?: (data: LlmUsage) => void
}

/** 会话上下文用量（估算值，用于聊天页「快超了」预警）。 */
export interface ContextUsage {
  used_tokens: number
  limit_tokens: number
  message_count: number
  model: string
}

/** 读取某会话的上下文用量（后端按会话全部消息估算）。 */
export async function getContextUsage(sessionId: string): Promise<ContextUsage> {
  const resp = await fetch(`${BASE}/session/${sessionId}/context`)
  if (!resp.ok) throw new Error(await errorText(resp))
  return resp.json()
}

/**
 * 截断会话：删除第 `fromIndex` 条消息及其之后的**全部**内容
 * （含随后产生的用户消息、AI 回复与工具记录），返回截断后的完整会话。
 *
 * 用于清掉发错或跑偏的消息，避免它污染后续上下文。
 */
export async function truncateSession(
  sessionId: string,
  fromIndex: number,
): Promise<AgentSession> {
  const resp = await fetch(`${BASE}/session/${sessionId}/truncate`, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ from_index: fromIndex }),
  })
  if (!resp.ok) throw new Error(await errorText(resp))
  return resp.json()
}

/** 滚动摘要内容（固定字段，与后端 `domain::session_summary::SessionSummary` 对齐）。 */
export interface SessionSummary {
  story_state: string
  confirmed: string[]
  open_threads: string[]
  next_step: string
}

/** 带元信息的滚动摘要（项目级恒定一份）。 */
export interface StoredSessionSummary {
  content: SessionSummary
  updated_at: string
}

/**
 * 读取当前会话所属项目的滚动摘要；从未收尾过时返回 `null`（不是错误）。
 */
export async function getSessionSummary(
  sessionId: string,
): Promise<StoredSessionSummary | null> {
  const resp = await fetch(`${BASE}/session/${sessionId}/summary`)
  if (!resp.ok) throw new Error(await errorText(resp))
  return resp.json()
}

/**
 * 会话收尾：让模型把这轮会话归纳成结构化摘要并**覆盖**项目那一份。
 *
 * 会连同「旧摘要」一起作为基线喂给模型，因此重复收尾不会丢掉前几轮的结论；
 * 生成后写进项目记忆，新建会话会自动读到。
 */
export async function summarizeSession(
  sessionId: string,
): Promise<StoredSessionSummary> {
  const resp = await fetch(`${BASE}/session/${sessionId}/summary`, {
    method: 'POST',
  })
  if (!resp.ok) throw new Error(await errorText(resp))
  return resp.json()
}

function parseSSEBlock(block: string): { event?: string; data?: string } {  let event: string | undefined
  let data = ''
  for (const line of block.split('\n')) {
    if (line.startsWith('event:')) event = line.slice(6).trim()
    else if (line.startsWith('data:')) data += line.slice(5).replace(/^ /, '')
  }
  return { event, data }
}

/**
 * 发送一条消息并消费 SSE 流。
 * 后端按命名事件下发：status(thinking) → 多个 token → done / error。
 *
 * `signal` 用于「停止生成」：中断后本函数会抛出 AbortError，由调用方识别为主动停止。
 */
export async function streamChat(
  sessionId: string,
  message: string,
  handlers: ChatStreamHandlers,
  signal?: AbortSignal,
): Promise<void> {
  const resp = await fetch(BASE + '/chat', {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ session_id: sessionId, message }),
    signal,
  })
  if (!resp.ok || !resp.body) {
    handlers.onError?.(`HTTP ${resp.status}`)
    return
  }

  const reader = resp.body.getReader()
  const decoder = new TextDecoder()
  let buffer = ''

  const flushBlock = (block: string) => {
    const ev = parseSSEBlock(block)
    if (!ev.event || ev.data === undefined) return
    if (ev.event === 'status') handlers.onStatus?.(ev.data)
    else if (ev.event === 'token') handlers.onToken?.(ev.data)
    else if (ev.event === 'done') handlers.onDone?.()
    else if (ev.event === 'error') handlers.onError?.(ev.data)
    else if (ev.event === 'usage') {
      try {
        handlers.onUsage?.(JSON.parse(ev.data) as LlmUsage)
      } catch (e) {
        // 用量是辅助信息，解析失败不应打断对话；但要留下痕迹便于排查
        console.warn('[agent] 用量事件解析失败:', e, ev.data)
      }
    } else if (ev.event === 'question') {
      try {
        handlers.onQuestion?.(JSON.parse(ev.data))
      } catch {
        // 解析失败忽略
      }
    } else if (ev.event === 'tool') {
      try {
        handlers.onTool?.(JSON.parse(ev.data))
      } catch {
        // 解析失败忽略
      }
    }
  }

  while (true) {
    const { done, value } = await reader.read()
    if (done) break
    buffer += decoder.decode(value, { stream: true })
    let idx: number
    while ((idx = buffer.indexOf('\n\n')) !== -1) {
      const block = buffer.slice(0, idx)
      buffer = buffer.slice(idx + 2)
      flushBlock(block)
    }
  }
  // 处理末尾可能未以 \n\n 结尾的残留块
  if (buffer.trim()) flushBlock(buffer)
}

File: crates/narrative-engine/src/api/agent.rs
//! Agent HTTP handlers（P1：/api/v1/agent/*）
//!
//! 对应 GPT 方案：创建 Session、工具列表、SSE 聊天、工具执行。
//! SSE 为**真实 token 级流式**：AgentRuntime 直接透传 LLM 的 delta，
//! 选择题则通过 `question` 事件下发（JSON：question + options）。

use axum::extract::{Path, Query, State, Json};
use axum::response::sse::{Event, KeepAlive, Sse};
use futures::Stream;
use serde::Deserialize;
use std::convert::Infallible;
use uuid::Uuid;

use crate::api::error::AppError;
use crate::state::AppState;
use agent::{
    ChatRequest, CreateSessionRequest, CreateSessionResponse, ExecuteToolRequest,
    ExecuteToolResponse, ListToolsResponse,
};
use serde_json;

/// `POST /api/agent/session` —— 创建引导会话。
pub async fn create_session(
    State(state): State<AppState>,
    Json(req): Json<CreateSessionRequest>,
) -> Result<Json<CreateSessionResponse>, AppError> {
    let id = state.agent.create_session(req.project_id).await?;
    Ok(Json(CreateSessionResponse { session_id: id }))
}

/// `GET /api/agent/session/{id}` —— 读取会话（含历史消息）。
pub async fn get_session(
    State(state): State<AppState>,
    Path(id): Path<Uuid>,
) -> Result<Json<agent::AgentSession>, AppError> {
    let session = state
        .agent
        .get_session(id)
        .await?
        .ok_or_else(|| anyhow::anyhow!("session not found: {}", id))?;
    Ok(Json(session))
}

/// `GET /api/v1/agent/session/{id}/context` —— 该会话的上下文用量（聊天页预警）。
pub async fn context_usage(
    State(state): State<AppState>,
    Path(id): Path<Uuid>,
) -> Result<Json<agent::ContextUsage>, AppError> {
    let usage = state.agent.context_usage(id).await?;
    Ok(Json(usage))
}

/// `GET /api/agent/sessions?project_id=xxx` —— 列出某项目下的会话（按项目隔离）。
pub async fn list_sessions(
    State(state): State<AppState>,
    Query(params): Query<ListSessionsQuery>,
) -> Result<Json<Vec<agent::AgentSession>>, AppError> {
    let project_id = Uuid::parse_str(&params.project_id)
        .map_err(|_| anyhow::anyhow!("Invalid project ID"))?;
    let sessions = state.agent.list_sessions_by_project(project_id).await?;
    Ok(Json(sessions))
}

/// `DELETE /api/agent/session/{id}` —— 删除会话（含消息，由存储层级联）。
pub async fn delete_session(
    State(state): State<AppState>,
    Path(id): Path<Uuid>,
) -> Result<(), AppError> {
    state.agent.delete_session(id).await?;
    Ok(())
}

/// `PUT /api/agent/session/{id}` —— 重命名会话。
pub async fn rename_session(
    State(state): State<AppState>,
    Path(id): Path<Uuid>,
    Json(req): Json<RenameSessionRequest>,
) -> Result<(), AppError> {
    state.agent.rename_session(id, &req.title).await?;
    Ok(())
}

#[derive(Debug, Deserialize)]
pub struct TruncateRequest {
    /// 从第几条消息开始删除（0 基下标）；该条及其之后的全部消息都会被移除。
    pub from_index: usize,
}

/// `POST /api/v1/agent/session/{id}/truncate` —— 截断会话（只保留上半部分）。
///
/// 删除第 `from_index` 条消息及其之后的全部内容（含后续的用户消息、AI 回复与工具记录），
/// 用于把发错或跑偏的消息连同其造成的上下文污染一起清掉。
/// 返回截断后的完整会话，便于前端直接替换界面。
pub async fn truncate_session(
    State(state): State<AppState>,
    Path(id): Path<Uuid>,
    Json(req): Json<TruncateRequest>,
) -> Result<Json<agent::AgentSession>, AppError> {
    let session = state.agent.truncate_session(id, req.from_index).await?;
    Ok(Json(session))
}

#[derive(Debug, Deserialize)]
pub struct RenameSessionRequest {
    title: String,
}

#[derive(Debug, Deserialize)]
pub struct ListSessionsQuery {
    project_id: String,
}

/// `GET /api/agent/tools` —— 列出当前可用工具。
pub async fn list_tools(
    State(state): State<AppState>,
) -> Result<Json<ListToolsResponse>, AppError> {
    Ok(Json(ListToolsResponse {
        tools: state.agent.list_tools(),
    }))
}

/// `POST /api/agent/tool/execute` —— 执行一个工具。
pub async fn execute_tool(
    State(state): State<AppState>,
    Json(req): Json<ExecuteToolRequest>,
) -> Result<Json<ExecuteToolResponse>, AppError> {
    let result = state.agent
        .execute_tool(req.project_id, &req.name, req.input)
        .await?;
    Ok(Json(ExecuteToolResponse {
        name: req.name,
        result,
    }))
}

/// `POST /api/agent/chat` —— SSE 聊天端点（真实 token 级流式）。
///
/// 先发 `status: thinking` 让客户端立即拿到 SSE 头；随后把 AgentRuntime 发出的
/// 流事件逐一转成 SSE 事件：`token`（文本片段）、`question`（选择题 JSON）、
/// `done`（结束）、`error`（出错）。
pub async fn chat(
    State(state): State<AppState>,
    Json(req): Json<ChatRequest>,
) -> Sse<impl Stream<Item = Result<Event, Infallible>>> {
    use async_stream::stream;
    use futures::StreamExt;
    let agent = state.agent.clone();
    let session_id = req.session_id;
    let message = req.message;

    let s = stream! {
        yield Ok::<_, Infallible>(Event::default().event("status").data("thinking"));
        let mut rs = match agent.chat_stream(session_id, &message).await {
            Ok(st) => st,
            Err(e) => {
                yield Ok(Event::default().event("error").data(e.to_string()));
                return;
            }
        };
        while let Some(ev) = rs.next().await {
            match ev {
                Ok(agent::AgentStreamEvent::Token(t)) => {
                    yield Ok(Event::default().event("token").data(t));
                }
                Ok(agent::AgentStreamEvent::Question { question, options }) => {
                    let payload = serde_json::json!({ "question": question, "options": options }).to_string();
                    yield Ok(Event::default().event("question").data(payload));
                }
                Ok(agent::AgentStreamEvent::Tool { name, input, ok, output }) => {
                    let payload = serde_json::json!({
                        "name": name,
                        "input": input,
                        "ok": ok,
                        "output": output,
                    })
                    .to_string();
                    yield Ok(Event::default().event("tool").data(payload));
                }
                Ok(agent::AgentStreamEvent::Usage(u)) => {
                    // 用量 + 缓存命中率：前端把它显示在「上下文」那一行，
                    // 用来判断网关侧提示缓存是否在正常工作（命中率长期为 0 说明每轮都在重算）。
                    let payload = serde_json::json!({
                        "prompt_tokens": u.prompt_tokens,
                        "completion_tokens": u.completion_tokens,
                        "total_tokens": u.total_tokens,
                        "cached_tokens": u.cached_tokens,
                        "cache_hit_rate": u.cache_hit_rate(),
                    })
                    .to_string();
                    yield Ok(Event::default().event("usage").data(payload));
                }
                Ok(agent::AgentStreamEvent::Done) => {
                    yield Ok(Event::default().event("done").data(""));
                }
                Ok(agent::AgentStreamEvent::Error(e)) => {
                    yield Ok(Event::default().event("error").data(e));
                }
                Err(e) => {
                    yield Ok(Event::default().event("error").data(e.to_string()));
                }
            }
        }
    };

    Sse::new(s).keep_alive(KeepAlive::default())
}

/// `GET /api/agent/prompt?scope=global` —— 读取当前生效提示词视图。
pub async fn get_prompt(
    State(state): State<AppState>,
    Query(params): Query<PromptQuery>,
) -> Result<Json<agent::PromptView>, AppError> {
    let scope = params.scope.unwrap_or_else(|| "global".to_string());
    let view = state.agent.get_prompt(&scope).await?;
    Ok(Json(view))
}

/// `PUT /api/agent/prompt` —— 保存（upsert）自定义提示词基座。
pub async fn save_prompt(
    State(state): State<AppState>,
    Json(req): Json<SavePromptRequest>,
) -> Result<Json<agent::PromptView>, AppError> {
    state
        .agent
        .save_prompt(&req.scope, &req.system_prompt, None)
        .await?;
    let view = state.agent.get_prompt(&req.scope).await?;
    Ok(Json(view))
}

/// `DELETE /api/agent/prompt?scope=global` —— 删除自定义覆盖（恢复内置默认）。
pub async fn delete_prompt(
    State(state): State<AppState>,
    Query(params): Query<PromptQuery>,
) -> Result<(), AppError> {
    let scope = params.scope.unwrap_or_else(|| "global".to_string());
    state.agent.delete_prompt(&scope).await?;
    Ok(())
}

#[derive(Debug, Deserialize)]
pub struct PromptQuery {
    scope: Option<String>,
}

#[derive(Debug, Deserialize)]
pub struct SavePromptRequest {
    scope: String,
    system_prompt: String,
}

/// `POST /api/v1/agent/guide/confirm` —— 前端用户点"确认推进"按钮触发。
///
/// 调 confirm_step 工具（已注册到 ToolRegistry），传入 project_id。
/// 工具内部完成：组装 MinCompleteSnapshot → 校验 → 通过则推进
/// `project.config.current_step` 到下一步。失败则返回结构化报告。
///
/// 可选参数 `target_step`：当用户从"血肉小选择器"点过来时传入，
/// 让后端校验指定 step 的产物（不传则按当前 stored_step 校验）。
pub async fn confirm_guide_step(
    State(state): State<AppState>,
    Json(req): Json<serde_json::Value>,
) -> Result<Json<serde_json::Value>, AppError> {
    let project_id = req
        .get("project_id")
        .and_then(|v| v.as_str())
        .and_then(|s| Uuid::parse_str(s).ok())
        .ok_or_else(|| anyhow::anyhow!("missing or invalid project_id"))?;
    let mut input = serde_json::json!({"project_id": project_id.to_string()});
    if let Some(ts) = req.get("target_step").and_then(|v| v.as_str()) {
        input["target_step"] = serde_json::json!(ts);
    }
    let result = state
        .agent
        .execute_tool(project_id, "confirm_step", input)
        .await?;
    Ok(Json(result))
}

/// `GET /api/v1/projects/{id}/guide/status` —— 引导进度的**只读真源**。
///
/// 前端步骤条的每一个对勾都取自这里，与 `confirm_step` 的推进校验共用同一套快照
/// 与同一个 `validate_step`，因此「界面打勾」与「点按钮能不能过」永远一致。
///
/// 存在理由：血肉步（地图 / 势力 / 道具 / 配角 / 副线）原先由前端各自数数量判断，
/// 一旦漏掉某一步，就会出现「产物明明齐了却永远不打勾」——而且没有任何报错。
pub async fn get_guide_status(
    State(state): State<AppState>,
    Path(project_id): Path<Uuid>,
) -> Result<Json<serde_json::Value>, AppError> {
    // 走已注册的 get_project_status 工具，而不是另写一份查询：
    // 前端打对勾的依据必须与 AI 回答、与 confirm_step 的校验是同一份判定。
    let input = serde_json::json!({"project_id": project_id.to_string()});
    let report = state
        .agent
        .execute_tool(project_id, "get_project_status", input)
        .await?;
    Ok(Json(report))
}

/// `POST /api/v1/agent/session/{id}/summary` —— 会话收尾：归纳成滚动摘要。
///
/// 把整段会话（连同项目已有的旧摘要作为基线）交给模型归纳成结构化 JSON，
/// **覆盖**项目那一份摘要，并同步一份到 `agent_memory` 供新会话注入。
/// 重复调用只会刷新那一份，不会堆积多份。
pub async fn summarize_session(
    State(state): State<AppState>,
    Path(id): Path<Uuid>,
) -> Result<Json<domain::session_summary::StoredSessionSummary>, AppError> {
    let summary = state.agent.summarize_session(id).await?;
    Ok(Json(summary))
}

/// `GET /api/v1/agent/session/{id}/summary` —— 读取该项目当前的滚动摘要。
///
/// 从未收尾过时返回 `null`（不是错误）：界面据此显示「还没有摘要」。
pub async fn get_session_summary(
    State(state): State<AppState>,
    Path(id): Path<Uuid>,
) -> Result<Json<Option<domain::session_summary::StoredSessionSummary>>, AppError> {
    let session = state
        .agent
        .get_session(id)
        .await?
        .ok_or_else(|| anyhow::anyhow!("session not found: {}", id))?;
    let summary = state.agent.load_summary(session.project_id).await?;
    Ok(Json(summary))
}
